### PR TITLE
Feature/contextual layers

### PR DESCRIPTION
**Pull request by Bluesmile82**
First PR for the contextual layers 🐎 
The map probably wont be needed. 
The layers are fetched from CARTO. The env has to be updated with the URL.
Created Tab, and imported switch buttons which can be reusable.
The Layers are updated in the contextual-layers microservice. There is an option for team layers for future use.
- [X] Add map with layers preview
- [X] Select and add GFW layers in the API
image

### CODE REVIEW

File: src/components/map/Map.js [hunk 1]
```diff
@@ -52,7 +56,19 @@ class Map extends React.Component {
                       .addTo(this.map)
                       .setZIndex(0);
   }
-
+    
+  updateLayers(layerUrls) {
+    layerUrls.forEach((layerUrl) => {
+      const lowCaseLayerUrl = layerUrl.replace("{Z}", "{z}").replace("{Y}", "{y}").replace("{X}", "{x}");
+      const layer = L.tileLayer(lowCaseLayerUrl);
+      if (!this.map.hasLayer(layer)) layer.addTo(this.map);
```
Comment (edbrett) [inline src/components/map/Map.js:64]: I know we aren't building the map now, but I think this should follow the pattern of all the other map parts be held in its own component. You can pass this.map up to the top level like in areas manage and pass the layers around that way.

File: src/components/pages/settings/LayersManager.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+import { includes } from '../../../helpers/utils';
+import Card from '../../ui/Card';
+import Tab from '../../ui/Tab';
+import SwitchButton from 'react-switch-button';
+import Checkbox from '../../ui/Checkbox';
+
+class LayersManager extends React.Component {
+  constructor() {
+    super();
+    this.state = {
```
Comment (edbrett) [inline src/components/pages/settings/LayersManager.js:14]: This should be at top level comp only.

File: src/modules/layers.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        };
+      }
+    }
+    default:
+      return state;
+  }
+}
+
+// Action Creators
+export function getGFWLayers() {
```
Comment (edbrett) [inline src/modules/layers.js:44]: For the sake of completeness, should we have some loading state here like other fetches? Might be useful when rendering the form and setting loaders etc.

File: src/modules/layers.js [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+}
+
+const parseLayer = function(layer){
+  return {
+    name: layer.title,
+    url: layer.tileurl
+  }
+}
+
+export function createLayer(layer, teamId) {
```
Comment (edbrett) [inline src/modules/layers.js:71]: Definitely need a loading prop action here.

File: src/modules/layers.js [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+        payload: {selectedLayer: normalized.contextualLayers}
+      });
+    })
+    .catch((error) => {
+      console.warn(error);
+    });
+  };
+}
+
+export function getLayers() {
```
Comment (edbrett) [inline src/modules/layers.js:101]: And here

File: src/styles/_react-switch-button.scss [hunk 6]
```diff
@@ -0,0 +1,229 @@
+/**
```
Comment (edbrett) [inline src/styles/_react-switch-button.scss:1]: Are these custom styles or from the npm module?

File: src/styles/components/_c-map.scss [hunk 7]
```diff
@@ -1,3 +1,9 @@
 .c-map {
   height: calc(100vh - #{$nav-height} - #{$hero-height} - #{$footer-height});
-}
+  &.-layers-container {
```
Comment (edbrett) [inline src/styles/components/_c-map.scss:3]: Not sure about this. Should be its own independant component placed separately. Like zoom controls etc.

**Review by edbrett — CHANGES_REQUESTED** [review]
Some comments for you. Also getting lots of bugs with selecting, deselecting, adding layers in general.

File: src/styles/_react-switch-button.scss (hunk 6 above)
Comment (Bluesmile82) [reply src/styles/_react-switch-button.scss:1]: They are custom. Copied from the module

**Comment by edbrett** [comment]
Closed as branched off by enable-layers

**Pull request closed without merging**